sklearn/feature_selection/sequential_feature_selector.py

```diff
@@ -0,0 +1,322 @@
+        - An int, giving the exact number of total jobs that are spawned
+        - A string, giving an expression as a function
+            of `n_jobs`, as in `2*n_jobs`
+
+    Attributes
+    ----------
+    estimator_ : scikit-learn Classifier or Regressor
+        Fitted clone of `estimator`.
+
+    feature_subset_idx_ : array-like, shape = [n_predictions]
+        Feature Indices of the selected feature subsets.
+
+    score_ : float
+        Cross validation average score of the selected subset.
+
+    subsets_ : dict
+        A dictionary of selected feature subsets during the
+        sequential selection, where the dictionary keys are
+        the lengths k of these feature subsets. The dictionary
+        values are dictionaries themselves with the following
+        keys: 'feature_subset_idx' (tuple of indices of the feature subset)
+              'cv_scores' (list individual cross-validation scores)
+              'avg_score' (average cross-validation score)
+
+    Examples
+    --------
+    The following example shows how to use the sequential feature selector
+    with default settings to select a feature subset, consisting of
+    1 to 3 features, from iris. The selection criteria for this
+    feature subset is the average cross-validation performance
+    (cv=5 by default) of the `estimator` (here: KNN)
+    during the greedy forward selection search.
+
+        >>> from sklearn.feature_selection import SequentialFeatureSelector
+        >>> from sklearn.neighbors import KNeighborsClassifier
+        >>> from sklearn.datasets import load_iris
+        >>> iris = load_iris()
+        >>> X, y = iris.data, iris.target
+        >>> knn = KNeighborsClassifier(n_neighbors=3)
+        >>> sfs = SequentialFeatureSelector(knn, n_features_to_select=(1, 3))
+        >>> sfs = sfs.fit(X, y)
+        >>> sfs.score_  # doctest: +ELLIPSIS
+        0.9733...
+        >>> sfs.feature_subset_idx_
+        (0, 2, 3)
+        >>> sfs.transform(X).shape
+        (150, 3)
+
+    """
+    def __init__(self, estimator, n_features_to_select=1,
+                 forward=True, scoring=None,
+                 cv=5, n_jobs=1,
+                 pre_dispatch='2*n_jobs'):
+
+        self.estimator = estimator
+        self.n_features_to_select = n_features_to_select
+        self.forward = forward
+        self.pre_dispatch = pre_dispatch
+        self.scoring = scoring
+        self.cv = cv
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """Perform feature selection and learn model from training data.
+
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+
+        """
+        if self.scoring is None:
```

Comment:
Oh nice, didn't know that! In this case, we can remove those lines, of course.

> specific reason for attaching the scorer in the instance itself?

It's used in several helper methods, so it would be easier to fetch the once instead of passing it around as function argument to the helper methods I guess. I renamed it to `self._scorer` but we can also completely remove it, just thought that it makes the code a bit more verbose